File: android-architecture-component/LatihanViewModel/java/app/src/main/java/com/dicoding/picodiploma/myviewmodel/MainActivity.java
package com.dicoding.picodiploma.myviewmodel;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.dicoding.picodiploma.myviewmodel.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding activityMainBinding;
    private MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityMainBinding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(activityMainBinding.getRoot());

        viewModel = new ViewModelProvider(this).get(MainViewModel.class);

        displayResult();

        activityMainBinding.btnCalculate.setOnClickListener(v -> {

            String width = activityMainBinding.edtWidth.getText().toString();
            String height = activityMainBinding.edtHeight.getText().toString();
            String length = activityMainBinding.edtLength.getText().toString();

            // Melakukan pengecekan apakah empty atau tidak
            if (width.isEmpty()) {
                activityMainBinding.edtWidth.setError("Masih kosong");
            } else if (height.isEmpty()) {
                activityMainBinding.edtHeight.setError("Masih kosong");
            } else if (length.isEmpty()) {
                activityMainBinding.edtLength.setError("Masih kosong");
            } else {

                viewModel.calculate(width, height, length);

                displayResult();
            }
        });
    }

    private void displayResult() {
        activityMainBinding.tvResult.setText(String.valueOf(viewModel.result));
    }
}

File: android-architecture-component/LatihanViewModel/kotlin/app/src/main/java/com/dicoding/picodiploma/myviewmodel/MainActivity.kt
package com.dicoding.picodiploma.myviewmodel

import android.os.Bundle
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import com.dicoding.picodiploma.myviewmodel.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var activityMainBinding: ActivityMainBinding
    private lateinit var viewModel: MainViewModel

//    jika menggunakan library activity-ktx
//    private val viewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)

        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)

        displayResult()

        activityMainBinding.btnCalculate.setOnClickListener {

            val width = activityMainBinding.edtWidth.text.toString()
            val height = activityMainBinding.edtHeight.text.toString()
            val length = activityMainBinding.edtLength.text.toString()

            when {
                width.isEmpty() -> {
                    activityMainBinding.edtWidth.error = "Masih kosong"
                }
                height.isEmpty() -> {
                    activityMainBinding.edtHeight.error = "Masih kosong"
                }
                length.isEmpty() -> {
                    activityMainBinding.edtLength.error = "Masih kosong"
                }
                else -> {

                    viewModel.calculate(width, height, length)

                    displayResult()
                }
            }
        }
    }

    private fun displayResult() {
        activityMainBinding.tvResult.text = viewModel.result.toString()
    }
}